### PR TITLE
doc/dev: add logging doc

### LINKED ISSUE #4870: doc: logging level guide
We should tell user how we choose logging level.

### PR DESCRIPTION
**Pull request by xiang90**
Fix #4870 

/cc @joshix

### CODE REVIEW

File: Documentation/dev/logging.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Logging Conventions
+
+We use [capnslog](capnslog) in etcd codebase for leveled logging. The following conventions for the log levels to use.
```
Comment (joshix) [inline Documentation/dev/logging.md]:
```
etcd uses the [capnslog][capnslog] library for logging application output categorized into *levels*. A log message's level is determined according to these conventions:
```

File: Documentation/dev/logging.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# Logging Conventions
+
+We use [capnslog](capnslog) in etcd codebase for leveled logging. The following conventions for the log levels to use. 
+
+  * Error: Data has been lost, a request has failed for a bad reason, or a required resource has been lost
+    * fail to allocate disk space for WAL
```
Comment (joshix) [inline Documentation/dev/logging.md]: Outdent this entire list by one level

Comment (joshix) [inline Documentation/dev/logging.md]: * Example: A failure to allocate disk space for WAL

File: Documentation/dev/logging.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  * Warning: (Hopefully) Temporary conditions that may cause errors, but may work fine. A replica disappearing (that may reconnect) is a warning.
+    * fail to send raft messages to a remote peer
```
Comment (joshix) [inline Documentation/dev/logging.md]: * Examples:
  * Failure to send raft message to a remote peer
  * Failure to recieve heartbeat message within the configured election timeout

**Comment by xiang90** [comment]
@joshix PTAL.

**Comment by joshix** [comment]
LGTM.

**Comment by mitake** [comment]
@xiang90 thanks for letting me know. I'll follow the guideline from next time.

**Pull request merged**